src/components/learnDetail/ScriptEdit.tsx

```diff
@@ -0,0 +1,109 @@
+import styled from 'styled-components';
+import { COLOR } from '@src/styles/color';
+interface ScriptType {
+  id: number;
+  text: string;
+  startTime: number;
+  endTime: number;
+}
+
+interface ScriptEditProps {
+  scripts: ScriptType[];
+  isHighlight: boolean;
+  isSpacing: boolean;
+}
+
+function ScriptEdit(props: ScriptEditProps) {
+  const { scripts, isHighlight, isSpacing } = props;
+
+  const handleClick = () => {
+    const selection = window.getSelection(); // 커서의 위치를 알 수 있음
+    const range = selection?.getRangeAt(0); // 커서의 startOffset과 endOffset을 갖고 있는 객체이다.
+    const startIdx = range?.startOffset; // 커서의 시작인덱스
+    // const endIdx = range?.endOffset; // 커서의 종료인덱스
+
+    // 선택한 텍스트가 빈칸인지 확인하는 로직
+    const selectedDiv = range?.startContainer as Node;
+    const serializer = new XMLSerializer();
+    const selectedLine = serializer.serializeToString(selectedDiv);
+
+    const isLeftBlank = startIdx && selectedLine[startIdx - 1] === ' '; //왼쪽이 빈칸
+    const isRightBlank = startIdx && selectedLine[startIdx] === ' '; //오른쪽이 빈칸
+    const isValidate = isLeftBlank || isRightBlank; //빈칸인지 여부
+
+    //클릭시 이벤트
+    if (selection?.type === 'Caret' && isValidate && isSpacing) {
+      const frag = document.createDocumentFragment();
+      const div = document.createElement('div');
+      isLeftBlank ? (div.innerHTML = '<span class=left >/</span>') : (div.innerHTML = '<span class=right >/</span>');
+      while (div.firstChild) {
+        frag.appendChild(div.firstChild);
+      }
+      range?.deleteContents();
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Range/deleteContents
deleteContents : The Range.deleteContents() method removes the contents of the Range from the Document